### ISSUE TITLE
New devices are not showing up in Devices Management after upgrade to 1.2.21

### ISSUE THREAD

**Issue by herr-nandez** [Cacti/cacti#4886]
# Describe the bug

When adding a new device (Management -> Devices -> Add) and saving them they do not appear in the Devices List. Changing filters has no effect. The device only exists in the database and can be edited by manually entering the device ID in the URL (host.php?action=edit&id=ENTERID).  When creating graphs for those devices the RRD files are not created and therefore no data is collected.
Existing devices are not affected and new graphs as well as existing ones are populated with data.

SELECT * FROM cacti.host LIMIT 0, 10000	
788 row(s) returned	0.031 sec / 0.063 sec

## SPINE Log:

```sh
SPINE: Using spine config file [/etc/spine.conf]
Total[0.0093] DEBUG: The path_php_server variable is /opt/data/www/cacti/script_server.php
Total[0.0093] DEBUG: The path_cactilog variable is /opt/data/www/cacti/log/cacti.log
Total[0.0093] DEBUG: The version variable is 8.0.29-commercial
Total[0.0093] DEBUG: The log_destination variable is 4 (STDOUT)
Total[0.0094] DEBUG: The path_php variable is /bin/php
Total[0.0095] DEBUG: The availability_method variable is 2
Total[0.0096] DEBUG: The ping_recovery_count variable is 3
Total[0.0097] DEBUG: The ping_failure_count variable is 2
Total[0.0098] DEBUG: The ping_method variable is 2
Total[0.0099] DEBUG: The ping_retries variable is 1
Total[0.0100] DEBUG: The ping_timeout variable is 400
Total[0.0101] DEBUG: The snmp_retries variable is 3
Total[0.0102] DEBUG: The log_perror variable is 1
Total[0.0103] DEBUG: The log_pwarn variable is 0
Total[0.0104] DEBUG: The boost_redirect variable is 1
Total[0.0105] DEBUG: The boost_rrd_update_enable variable is 1
Total[0.0106] DEBUG: The log_pstats variable is 0
Total[0.0107] DEBUG: The threads variable is 45
Total[0.0109] DEBUG: The polling interval is 60 seconds
Total[0.0110] DEBUG: The number of concurrent processes is 10
Total[0.0111] DEBUG: The script timeout is 10
Total[0.0112] DEBUG: The selective_device_debug variable is 
Total[0.0113] DEBUG: The spine_log_level variable is 0
Total[0.0114] DEBUG: The number of php script servers to run is 15
Total[0.0115] DEBUG: The number of active data source profiles is 1
Total[0.0116] DEBUG: The number of snmp ports on the system is 1
Total[0.0118] DEBUG: StartDevice='1201', EndDevice='1201', TotalPHPScripts='0'
Total[0.0119] DEBUG: The PHP Script Server is Not Required
Total[0.0120] DEBUG: The Maximum SNMP OID Get Size is 20
Total[0.0154] DEBUG: Total Connections made 1
Total[0.0155] DEBUG: Creating Local Connection Pool of 45 threads.
Total[0.0155] DEBUG: Creating Local Connection 0.
Total[0.0192] DEBUG: Total Connections made 2
Total[0.0192] DEBUG: Creating Local Connection 1.
Total[0.0221] DEBUG: Total Connections made 3
Total[0.0221] DEBUG: Creating Local Connection 2.
Total[0.0255] DEBUG: Total Connections made 4
Total[0.0256] DEBUG: Creating Local Connection 3.
Total[0.0297] DEBUG: Total Connections made 5
Total[0.0297] DEBUG: Creating Local Connection 4.
Total[0.0329] DEBUG: Total Connections made 6
Total[0.0329] DEBUG: Creating Local Connection 5.
Total[0.0360] DEBUG: Total Connections made 7
Total[0.0360] DEBUG: Creating Local Connection 6.
Total[0.0389] DEBUG: Total Connections made 8
Total[0.0389] DEBUG: Creating Local Connection 7.
Total[0.0429] DEBUG: Total Connections made 9
Total[0.0429] DEBUG: Creating Local Connection 8.
Total[0.0457] DEBUG: Total Connections made 10
Total[0.0457] DEBUG: Creating Local Connection 9.
Total[0.0486] DEBUG: Total Connections made 11
Total[0.0486] DEBUG: Creating Local Connection 10.
Total[0.0517] DEBUG: Total Connections made 12
Total[0.0517] DEBUG: Creating Local Connection 11.
Total[0.0546] DEBUG: Total Connections made 13
Total[0.0546] DEBUG: Creating Local Connection 12.
Total[0.0578] DEBUG: Total Connections made 14
Total[0.0578] DEBUG: Creating Local Connection 13.
Total[0.0610] DEBUG: Total Connections made 15
Total[0.0611] DEBUG: Creating Local Connection 14.
Total[0.0647] DEBUG: Total Connections made 16
Total[0.0648] DEBUG: Creating Local Connection 15.
Total[0.0675] DEBUG: Total Connections made 17
Total[0.0675] DEBUG: Creating Local Connection 16.
Total[0.0705] DEBUG: Total Connections made 18
Total[0.0705] DEBUG: Creating Local Connection 17.
Total[0.0733] DEBUG: Total Connections made 19
Total[0.0733] DEBUG: Creating Local Connection 18.
Total[0.0767] DEBUG: Total Connections made 20
Total[0.0767] DEBUG: Creating Local Connection 19.
Total[0.0795] DEBUG: Total Connections made 21
Total[0.0795] DEBUG: Creating Local Connection 20.
Total[0.0826] DEBUG: Total Connections made 22
Total[0.0827] DEBUG: Creating Local Connection 21.
Total[0.0857] DEBUG: Total Connections made 23
Total[0.0857] DEBUG: Creating Local Connection 22.
Total[0.0886] DEBUG: Total Connections made 24
Total[0.0887] DEBUG: Creating Local Connection 23.
Total[0.0916] DEBUG: Total Connections made 25
Total[0.0916] DEBUG: Creating Local Connection 24.
Total[0.0945] DEBUG: Total Connections made 26
Total[0.0945] DEBUG: Creating Local Connection 25.
Total[0.0972] DEBUG: Total Connections made 27
Total[0.0972] DEBUG: Creating Local Connection 26.
Total[0.0998] DEBUG: Total Connections made 28
Total[0.0998] DEBUG: Creating Local Connection 27.
Total[0.1028] DEBUG: Total Connections made 29
Total[0.1028] DEBUG: Creating Local Connection 28.
Total[0.1058] DEBUG: Total Connections made 30
Total[0.1059] DEBUG: Creating Local Connection 29.
Total[0.1089] DEBUG: Total Connections made 31
Total[0.1089] DEBUG: Creating Local Connection 30.
Total[0.1117] DEBUG: Total Connections made 32
Total[0.1117] DEBUG: Creating Local Connection 31.
Total[0.1144] DEBUG: Total Connections made 33
Total[0.1144] DEBUG: Creating Local Connection 32.
Total[0.1171] DEBUG: Total Connections made 34
Total[0.1171] DEBUG: Creating Local Connection 33.
Total[0.1199] DEBUG: Total Connections made 35
Total[0.1200] DEBUG: Creating Local Connection 34.
Total[0.1228] DEBUG: Total Connections made 36
Total[0.1228] DEBUG: Creating Local Connection 35.
Total[0.1258] DEBUG: Total Connections made 37
Total[0.1258] DEBUG: Creating Local Connection 36.
Total[0.1287] DEBUG: Total Connections made 38
Total[0.1288] DEBUG: Creating Local Connection 37.
Total[0.1315] DEBUG: Total Connections made 39
Total[0.1316] DEBUG: Creating Local Connection 38.
Total[0.1346] DEBUG: Total Connections made 40
Total[0.1347] DEBUG: Creating Local Connection 39.
Total[0.1381] DEBUG: Total Connections made 41
Total[0.1381] DEBUG: Creating Local Connection 40.
Total[0.1411] DEBUG: Total Connections made 42
Total[0.1411] DEBUG: Creating Local Connection 41.
Total[0.1437] DEBUG: Total Connections made 43
Total[0.1437] DEBUG: Creating Local Connection 42.
Total[0.1465] DEBUG: Total Connections made 44
Total[0.1465] DEBUG: Creating Local Connection 43.
Total[0.1490] DEBUG: Total Connections made 45
Total[0.1490] DEBUG: Creating Local Connection 44.
Total[0.1526] DEBUG: Total Connections made 46
Total[0.1527] DEBUG: Version 1.2.21 starting
Total[0.1527] DEBUG: MySQL is Thread Safe!
Total[0.1527] DEBUG: Spine running as 2645 UID, 0 EUID
Total[0.1528] DEBUG: Spine is running as root.
Total[0.1528] DEBUG: Spine has got ICMP
Total[0.1529] DEBUG: Initializing Net-SNMP API
Total[0.1529] DEBUG: Issues with SNMP Header Version information, assuming old version of Net-SNMP.
Total[0.1551] DEBUG: Initializing PHP Script Server(s)
Total[0.1572] Spine will support multithread device polling.
Total[0.1574] DEBUG: Initial Value of Available Threads is 45 (0 outstanding)
Total[0.1576] DEBUG: Valid Thread to be Created (140278690002688)
Total[0.1576] DEBUG: Available Threads is 44 (1 outstanding)
Total[0.1577] DEBUG: Device[0] HT[1] In Poller, About to Start Polling
Total[0.1579] DEBUG: Valid Thread to be Created (140278597940992)
Total[0.1579] DEBUG: Available Threads is 43 (2 outstanding)
Total[0.1579] DEBUG: Traversing Local Connection Pool for free connection.
Total[0.1579] WARNING: Device[1201] polling sleeping while waiting for 2 Threads to End
Total[0.1580] DEBUG: Device[1201] HT[1] In Poller, About to Start Polling
Total[0.1580] DEBUG: Checking Local Pool ID 0.
Total[0.1580] DEBUG: Allocating Local Pool ID 0.
Total[0.1581] DEBUG: Traversing Local Connection Pool for free connection.
Total[0.1581] DEBUG: Checking Local Pool ID 0.
Total[0.1581] DEBUG: Checking Local Pool ID 1.
Total[0.1581] DEBUG: Allocating Local Pool ID 1.
Total[0.1584] DEBUG: Freeing Local Pool ID 1
Total[0.1591] Device[0] HT[1] Total Time: 0.0013 Seconds
Total[0.1593] DEBUG: Freeing Local Pool ID 0
Total[0.1594] DEBUG: Device[0] HT[1] DEBUG: HOST COMPLETE: About to Exit Device Polling Thread Function
Total[0.6581] The final count of Threads is 0
Total[0.6581] INFO: Thread complete for Device[0] and 0 to 0 sources
Total[0.6582] INFO: Thread incomplete for Device[1201] and 0 to 0 sources
Total[0.6582] DEBUG: Closing Local Connection Pool ID 0
Total[0.6584] DEBUG: Closing Local Connection Pool ID 1
Total[0.6585] DEBUG: Closing Local Connection Pool ID 2
Total[0.6587] DEBUG: Closing Local Connection Pool ID 3
Total[0.6588] DEBUG: Closing Local Connection Pool ID 4
Total[0.6588] DEBUG: Closing Local Connection Pool ID 5
Total[0.6588] DEBUG: Closing Local Connection Pool ID 6
Total[0.6589] DEBUG: Closing Local Connection Pool ID 7
Total[0.6589] DEBUG: Closing Local Connection Pool ID 8
Total[0.6590] DEBUG: Closing Local Connection Pool ID 9
Total[0.6590] DEBUG: Closing Local Connection Pool ID 10
Total[0.6590] DEBUG: Closing Local Connection Pool ID 11
Total[0.6591] DEBUG: Closing Local Connection Pool ID 12
Total[0.6591] DEBUG: Closing Local Connection Pool ID 13
Total[0.6592] DEBUG: Closing Local Connection Pool ID 14
Total[0.6592] DEBUG: Closing Local Connection Pool ID 15
Total[0.6593] DEBUG: Closing Local Connection Pool ID 16
Total[0.6593] DEBUG: Closing Local Connection Pool ID 17
Total[0.6594] DEBUG: Closing Local Connection Pool ID 18
Total[0.6594] DEBUG: Closing Local Connection Pool ID 19
Total[0.6595] DEBUG: Closing Local Connection Pool ID 20
Total[0.6595] DEBUG: Closing Local Connection Pool ID 21
Total[0.6595] DEBUG: Closing Local Connection Pool ID 22
Total[0.6596] DEBUG: Closing Local Connection Pool ID 23
Total[0.6596] DEBUG: Closing Local Connection Pool ID 24
Total[0.6597] DEBUG: Closing Local Connection Pool ID 25
Total[0.6597] DEBUG: Closing Local Connection Pool ID 26
Total[0.6598] DEBUG: Closing Local Connection Pool ID 27
Total[0.6598] DEBUG: Closing Local Connection Pool ID 28
Total[0.6599] DEBUG: Closing Local Connection Pool ID 29
Total[0.6599] DEBUG: Closing Local Connection Pool ID 30
Total[0.6599] DEBUG: Closing Local Connection Pool ID 31
Total[0.6600] DEBUG: Closing Local Connection Pool ID 32
Total[0.6600] DEBUG: Closing Local Connection Pool ID 33
Total[0.6601] DEBUG: Closing Local Connection Pool ID 34
Total[0.6601] DEBUG: Closing Local Connection Pool ID 35
Total[0.6602] DEBUG: Closing Local Connection Pool ID 36
Total[0.6602] DEBUG: Closing Local Connection Pool ID 37
Total[0.6602] DEBUG: Closing Local Connection Pool ID 38
Total[0.6603] DEBUG: Closing Local Connection Pool ID 39
Total[0.6603] DEBUG: Closing Local Connection Pool ID 40
Total[0.6604] DEBUG: Closing Local Connection Pool ID 41
Total[0.6604] DEBUG: Closing Local Connection Pool ID 42
Total[0.6605] DEBUG: Closing Local Connection Pool ID 43
Total[0.6605] DEBUG: Closing Local Connection Pool ID 44
Total[0.6605] DEBUG: Thread Cleanup Complete
Total[0.6606] DEBUG: PHP Script Server Pipes Closed
Total[0.6606] DEBUG: Allocated Variable Memory Freed
Total[0.6606] DEBUG: MYSQL Free & Close Completed
Total[0.6607] DEBUG: Net-SNMP Close Completed
Total[0.6607] Time: 0.6606 s, Threads: 45, Devices: 2
```

## System

- OS: RHEL 7.9
- Cacti: Version 1.2.21
- Spine: Version 1.2.21
- RRDTool: 1.4.8
- Net-SNMP: v5.7.2
- Polling: 60 Seconds
- Structured RRA Paths

## Screenshots
![device](https://user-images.githubusercontent.com/48107295/184360533-2fe70e03-7f00-4564-bfe5-f21f6e199e83.png)
![hosts](https://user-images.githubusercontent.com/48107295/184360536-6333a30d-160a-4410-8d0d-f565a17a9f64.png)

**Comment by TheWitness** [comment]
I fixed your spine output to make it readable.

**Comment by TheWitness** [comment]
You should also run the cli:

```
cd /var/www/html/cacti/cli
php -q audit_database.php --report
```

Post any exceptions.

**Comment by TheWitness** [comment]
What MariaDB/MySQL version are you using?  It looks like there is some problem there.  Also, for the Host query, you need to replace the `?` with the `device_id`.

**Comment by TheWitness** [comment]
```sql
ALTER TABLE host MODIFY COLUMN deleted CHAR(2) NOT NULL default '';
```

Not that you are going to likely have some issues with this MySQL version.  For example, the audit script was having some issues that we were unaware of.  Keep that in mind.  Oracle is breaking compatibility all over the place.

**Comment by TheWitness** [comment]
Related to https://github.com/Cacti/cacti/issues/4887

**Comment by TheWitness** [comment]
This will be resolved in 1.2.22 by #4887.  For now, just do that alter statement and you will be back in business.  I'll be opening a new ticket for additional MySQL 8.x testing and certification.  Right now, it's not certified (but most things 'should' work).
